pynumaflow/reducestreamer/servicer/async_servicer.py

```diff
@@ -111,22 +111,35 @@ async def ReduceFn(
         try:
             async for msg in consumer:
                 # If the message is an exception, we raise the exception
-                if isinstance(msg, Exception):
-                    await handle_error(context, msg)
-                    raise msg
+                if isinstance(msg, BaseException):
+                    handle_error(context, msg)
```

Comment:
why don't i see `handle_error` in a few other places?

Reply:
`handle_error` is used in reducestreamer only as in this function we extract the traceback from the exception before logging it out. 
In other places we log out the exception along with the traceback option directly.

But in reducestreamer we write the exceptions to the global result queue, hence while processing them we extract the traceback and then log it for better contextual logging.